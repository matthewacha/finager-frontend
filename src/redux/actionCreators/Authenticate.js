import {
  USER_LOGIN,
  GOOGLE_AUTH,
//   GOOGLE_AUTH_SUCCESS,
  GOOGLE_AUTH_FAILURE,
  USER_LOGIN_FAILURE,
  USER_LOGIN_SUCCESS
} from "../actionTypes";

export const loginUser = (data) => ({
  type: USER_LOGIN,
  data
});

export const loginUserFailure = (error) => ({
  type: USER_LOGIN_FAILURE,
  error
});

export const loginUserSuccess = (data) => ({
  type: USER_LOGIN_SUCCESS,
  data
});

export const googleAuth = () => ({
  type: GOOGLE_AUTH
});

// export const googleAuthSuccess = () => ({
//     type: GOOGLE_AUTH_SUCCESS
//   });

export const googleAuthFailure = (error) => ({
    type: GOOGLE_AUTH_FAILURE,
    error
  });
