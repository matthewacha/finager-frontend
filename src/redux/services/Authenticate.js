import axios from "axios";

const baseUrl = process.env.FINAGER_APP_URL;

class AuthenticationApi {
    static googleAuth(){
        // process.env.GOOGLE_APP_LOGIN_API
        const error = new Error('Wrong credentials')
        error.error = {
            status_code: 401,
            error: 'Wrong credentials'
        }
        const success = {
            status_code: 200,
            data: {id_token: 'eyTggbjkikjf6ft7lm0jjm',
                    email: 'testing@mail.com'}
        } 
        // throw error;
        return success;
    }

    static loginUser(data){
        
        return data;
    }
}

export default AuthenticationApi;