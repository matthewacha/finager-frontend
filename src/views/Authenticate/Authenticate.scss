
// .auth__panel--social:hover {
//     cursor: pointer;
//     // background-color: #c8c8c840;
//     box-shadow: rgba(0, 0, 0, 0.2) 0px 1rem 1rem -1rem;
//     transition: background-color 0.5s, box-shadow 0.2s ease-out;
// }

// .auth__panel--video:hover {
//     cursor: pointer;
//     background-color: #c8c8c840;
//     transition: background-color 0.2s ease-out;
// }

// .auth__panel--try:hover {
//     cursor: pointer;
//     background-color: #c8c8c840;
//     transition: background-color 0.2s ease-out;
// }

// .auth__panel--icon:hover {
//     cursor: pointer;
//     box-shadow: rgba(0, 0, 0, 0.2) 0px 1rem 1rem -1rem;
//     transition: box-shadow 0.2s linear 0s;
// }

// .auth__panel--terms:hover {
//     cursor: pointer;
//     color: #016682;
//     text-decoration: underline solid #016682;
// }

.login-background{
    height: 840px;
    width: 100%;
    display: inline-flex;
    @media screen and (min-width: 760px) and (max-width: 1024px) {
        display: block;
    }
    .auth{
        &__left-column{
            width: 40%;
            height: inherit;
            @media screen and (max-width: 550px){
                width: 100%;
            }
        }
        &__right-column{
            background-color: #016682eb;
            width: 60%;
            height: inherit;
            overflow-x: scroll;
            @media screen and (max-width: 550px){
                width: 0%;
                height: inherit;
                visibility: collapse;
            }
            &__panel{
                margin: auto;
                padding: 10%;
                width: auto;
                height: 39%;
                background: #737f82;
                &--text{
                    position: relative;
                    top: 4px;
                    color: aliceblue;
                    font-family: 'Tahoma';
                    font-size: 20px;
                    left: 15px;
                    display: flex;
                }
            }
        }
        &__panel{
            width: 100%;
            height: inherit;
            border-radius: 4px;
            position: relative;
            &--icon{
                margin: auto;
                margin-top: 10%;
                width: 50%;
                height: 220%;
                border: 1px solid #97938f40;
                border-radius: 4px;
                position: relative;
                z-index: 2;
                background: white;
                &__back{
                    position: relative;
                    z-index: 1;
                    padding: 10%;
                    padding-top:0px;
                    width: auto;
                    height: 15%;
                    border: 1px solid #00000000;
                    background: #016682;
                }
                .icon-image{
                    width: -webkit-fill-available !important;
                    height: -webkit-fill-available !important;
                }
            }
            &--icon:hover {
                cursor: pointer;
                box-shadow: rgba(0, 0, 0, 0.2) 0px 1rem 1rem -1rem;
                transition: box-shadow 0.2s linear 0s;
            }

            &--text{
                padding: 10%;
                width: auto;
                height: 15%;
            }
            &--intermediate{
                padding: 10%;
                width: auto;
                height: 20%;
                position: relative;
                z-index: -2;
            }
            &--social{
                display: inline-flex;
                top:100px;
                left: 100px;
                margin: 2%;
                margin-top: 5%;
                margin-left: 21%;
                padding: 10px;
                width: 50%;
                height: 8%;
                border: 1px solid black;
                border-radius: 30px;
                border: 1px solid #97938f30;
            }
            &--social:hover {
                cursor: pointer;
                // background-color: #c8c8c840;
                box-shadow: rgba(0, 0, 0, 0.2) 0px 1rem 1rem -1rem;
                transition: background-color 0.5s, box-shadow 0.2s ease-out;
            }
            &--video{
                display: inline-flex;
                position: relative;
                top: 80px;
                left: 25px;
                margin-left: 11%;
                padding: 10px;
                width: 26%;
                height: 30px;
                border-radius: 98px;
                border: 2px solid #ffffff9c;
            }
            &--video:hover {
                cursor: pointer;
                background-color: #c8c8c840;
                transition: background-color 0.2s ease-out;
            }
            
            &--try{
                display: inline-flex;
                position: relative;
                top: 80px;
                left: 25px;
                margin-left: 11%;
                padding: 10px;
                width: 26%;
                height: 30px;
                border-radius: 98px;
                border: 2px solid #ffffff9c;
            }

            &--try:hover {
                cursor: pointer;
                background-color: #c8c8c840;
                transition: background-color 0.2s ease-out;
            }

            &--terms{
                margin-left: 21%;
                font-size: 15px;
            }
            &--terms:hover {
                cursor: pointer;
                color: #016682;
                text-decoration: underline solid #016682;
            }

            &--foot{
                margin-top: 23.1%;
                padding: 2%;
                width: auto;
                height: 3%;
                border: 1px solid #016682;
                background: #016682;
            }
        }
    }

}

.google-icon{
    width: fit-content;
    @media screen and (max-width: 550px){
        width: 34px;
    }
}

.video-icon{
    width: fit-content;
    position: relative;
    left: -2px;
    @media screen and (max-width: 550px){
        width: 34px;
    }
}

.auth-text{
    color: #737f82;
    font-family: 'Tahoma';
    font-size: 25px;
    align-self: center;
}

.side-button__text{
    color: #ffffff9c;
    font-family: 'Tahoma';
    font-size: 22px;
    position: relative;
    left: -17px;
    align-self: center;
    &--try{
        position: relative;
        left: 84px;
        align-self: center;
    }
}