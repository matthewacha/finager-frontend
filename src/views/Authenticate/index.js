import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import "./Authenticate.scss";

import googleIcon from "../../assets/images/googleIcon.svg";
import Button from "../../components/Button/Button";
import {
  loginUser,
  loginUserFailure,
  googleAuthFailure
} from "../../redux/actionCreators/Authenticate";


class Authentication extends Component {
  state = {
    status: "signup"
  };

  handleLoad = (successCallback, errorCallback) => {
    const params = {
      apiKey: "AIzaSyDhTzY6EjccIISxr7A0j_RhLkGkdCiAig4",
      discoveryDocs: [
        "https://people.googleapis.com/$discovery/rest?version=v1"
      ],
      clientId:
        "599280857870-jubqh41bgsolgejejsfvalhqinu092mt.apps.googleusercontent.com",
      scope: "profile"
    };
    window.gapi.load("auth2", () => {
      // initialize
      window.gapi.auth2.init(params).then(
        res => {
          // listen for sign in state changes
          // window.gapi.auth2.getAuthInstance().isSignedIn.listen(this.updateSigninStatus);
          if (res.isSignedIn.get()) {
            // handle success action creator here
            successCallback(res.currentUser.get());
            console.log(res.currentUser.get());
          }
        },
        // handle error action creator err =>
        error => {
          errorCallback(error);
        }
      );
    });
  };

  loadGoogleClient = callback => {
    //   Load the google API client and auth2 library
    const script = document.createElement("script");
    script.src = "https://apis.google.com/js/api.js";
    // add script element to the DOM
    document.body.appendChild(script);
    script.onload = callback;
  };

  componentDidMount() {
    const { loginUser, googleAuthFailure } = this.props;
    this.loadGoogleClient(this.handleLoad(loginUser, googleAuthFailure));
  }

  handleClick = event => {
    event.preventDefault();
    const { googleAuth, successCallback, googleAuthFailure } = this.props;
    const auth = window.gapi.auth2.getAuthInstance();
    auth.signIn().then(
      res => {
        // handle success action creator here
        successCallback(res);
        console.log(res);
      },
      err => {
        // handle error action creator here
        googleAuthFailure(err);
      }
    );
  };

  handleTerms = event => {
    event.preventDefault();
    console.log("Redirect to terms...");
  };

  handleTryOut = event => {
    event.preventDefault();
    console.log("Trying out app...");
  };

  handleVideo = event => {
    event.preventDefault();
    console.log("Redirect to YouTube...");
  };

  render() {
    return (
      <div className={"login-background"}>
        <div className={"auth__left-column"}>
          <div type="presentation" className={"auth__panel"}>
            <div type="presentation" className={"auth__panel--icon__back"}>
              <div type="presentation" className={"auth__panel--icon"}>
                <img
                  src={googleIcon}
                  alt={"google-icon"}
                  className={"icon-image"}
                />
              </div>
            </div>
            <div type="presentation" className={"auth__panel--intermediate"} />
            <Button
              hasIcon={true}
              iconAlt={"google-icon"}
              iconStyle={"google-icon"}
              iconImage={googleIcon}
              customButtonStyle={"auth__panel--social"}
              disable={"false"}
              text={"Login with Google"}
              textStyle={"auth-text"}
              onClick={this.handleClick}
              hasText={true}
            />
            <div
              role="button"
              onClick={this.handleTerms}
              className={"auth__panel--terms"}
            >
              Clicking login confirms that you agree to our terms and
              conditions.
            </div>

            <div type="presentation" className={"auth__panel--foot"} />
          </div>
        </div>
        <div className={"auth__right-column"}>
          <div type="presentation" className={"auth__right-column__panel"} />
          <div className={"auth__right-column__panel--text"}>
            <Button
              hasIcon={true}
              iconAlt={"video"}
              iconStyle={"video-icon"}
              iconImage={googleIcon}
              customButtonStyle={"auth__panel--video"}
              disable={"false"}
              text={"Watch Tutorial"}
              textStyle={"side-button__text"}
              onClick={this.handleVideo}
              hasText={true}
            />
            <Button
              hasIcon={false}
              customButtonStyle={"auth__panel--try"}
              disable={"false"}
              text={"Try it out"}
              textStyle={"side-button__text--try"}
              onClick={this.handleTryOut}
              hasText={true}
            />
          </div>
        </div>
      </div>
    );
  }
}

Authentication.propTypes = {
  loginUser: PropTypes.func,
  googleAuthFailure: PropTypes.func,
  loginUserFailure: PropTypes.func
};

Authentication.defaultProps = {
  loginUser: () => {},
  googleAuthFailure: () => {},
  loginUserFailure: () => {}
};

const mapStateToProps = ({ auth }) => ({
  error: auth.error
});

const actionCreators = {
  googleAuthFailure,
  loginUserFailure,
  loginUser
};

export default connect(
  mapStateToProps,
  actionCreators
)(Authentication);
