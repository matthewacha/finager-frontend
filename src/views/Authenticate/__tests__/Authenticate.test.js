/* eslint-disable react/react-in-jsx-scope */
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Authenticate from '../index';

const props = {
    loginUser: jest.fn(),
}

const mockStore = configureStore();

const initialStore = {

};

const store = mockStore(initialStore);

describe('Test Authenticate component', () => {
    it('Test it renders correctlt', () => {
        const wrapper = shallow(
            <Provider store={store} >
                <Authenticate store={store} props={props} />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    })
})