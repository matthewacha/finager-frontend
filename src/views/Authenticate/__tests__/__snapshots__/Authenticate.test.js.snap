// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Authenticate component Test it renders correctlt 1`] = `
<Connect(Authentication)
  props={
    Object {
      "loginUser": [MockFunction],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
